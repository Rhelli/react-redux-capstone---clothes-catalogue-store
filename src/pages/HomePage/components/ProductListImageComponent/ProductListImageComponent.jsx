import React from 'react';
import PropTypes from 'prop-types';

const ProductListImageComponent = ({ images }) => (
  <div>
    <div>
      <img alt="" src={images} />
    </div>
  </div>
);

ProductListImageComponent.propTypes = {
  images: PropTypes.string.isRequired,
};

export default ProductListImageComponent;
