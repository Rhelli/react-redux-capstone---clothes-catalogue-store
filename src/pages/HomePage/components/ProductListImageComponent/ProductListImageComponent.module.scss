.productListImage {
  width: 100%;

  img {
    width: 100%;
    border-radius: 15px;
  }
}
