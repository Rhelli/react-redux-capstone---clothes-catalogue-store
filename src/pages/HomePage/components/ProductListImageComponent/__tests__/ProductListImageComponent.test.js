import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ProductListImageComponent from '../ProductListImageComponent';

configure({ adapter: new Adapter() });

const images = 'thisIsAnImageURL';
const productListImageComponent = <ProductListImageComponent images={images} />;

describe('The appearance of the ProductListImageComponent', () => {
  it('Matches the ProductListImageComponent snapshot', () => {
    const tree = renderer.create(productListImageComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Has only one child, with the correct className', () => {
    const { container } = render(productListImageComponent);
    expect(container.childElementCount).toEqual(1);
    expect(container.firstChild.nodeName).toBe('DIV');
    expect(container.firstChild.classList.contains('productListImage'));
  });

  it('Contains a single image tag which contains a non-null src value', () => {
    const { container } = render(productListImageComponent);
    expect(container.firstChild.firstChild.nodeName).toBe('IMG');
    expect(container.firstChild.firstChild).toHaveAttribute('src', 'thisIsAnImageURL');
  });
});
