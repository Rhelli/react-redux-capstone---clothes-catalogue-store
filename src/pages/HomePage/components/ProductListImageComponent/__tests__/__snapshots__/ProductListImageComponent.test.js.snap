// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The appearance of the ProductListImageComponent Matches the ProductListImageComponent snapshot 1`] = `
<div
  className="productListImage"
>
  <img
    alt=""
    src="thisIsAnImageURL"
  />
</div>
`;
